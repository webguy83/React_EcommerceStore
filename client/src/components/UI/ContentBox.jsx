import React from 'react';
import styled from 'styled-components/macro';

const Content = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: .1rem solid var(--prim);
    background-color: white;
    padding: 1.3rem;
    opacity: .6;

    & > h2 {
        text-transform: uppercase;
    }
    & > p {
        font-size: 1.3rem;
    }
    ${({ style }) => style}
`

const ContentBox = ({ title, textContent, ...props }) => {
    return (
        <Content className="content" {...props}>
            <h2>{title}</h2>
            <p>{textContent}</p>
        </Content>
    );
};

export default ContentBox;