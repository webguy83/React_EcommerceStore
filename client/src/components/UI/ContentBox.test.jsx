import React from "react";
import ContentBox from "./ContentBox";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

describe("content box tests", () => {
  let container = null;

  beforeEach(() => {
    container = render(
      <ContentBox title="Canada" textContent="Shop for Photos!" />
    );
  });

  it("should render without crashing", () => {
    const { asFragment } = container;
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render heading text", () => {
    const { getByTestId } = container;
    expect(getByTestId("title")).toHaveTextContent(/^Canada$/);
  });

  it("should render text content text", () => {
    const { getByTestId } = container;
    expect(getByTestId("textContent")).toHaveTextContent(/^Shop for Photos!$/);
  });
});

// describe("css testing", () => {
//   it("should add new custom CSS to styled component", () => {
//     const { getByTestId } = render(
//       <ContentBox
//         title="Canada"
//         textContent="Shop for Photos!"
//         style={{
//           fontWeight: "bold"
//         }}
//       />
//     );
//     expect(getByTestId("content")).toHaveStyleRule("font-weight", "bold");
//   });
// });
