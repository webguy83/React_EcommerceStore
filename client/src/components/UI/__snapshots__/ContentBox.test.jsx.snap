// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content box tests should render without crashing 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.1rem solid var(--prim);
  background-color: white;
  padding: 1.3rem;
  opacity: 0.6;
}

.c0 > h2 {
  text-transform: uppercase;
}

.c0 > p {
  font-size: 1.3rem;
}

<div
    class="c0 content"
    data-testid="content"
  >
    <h2
      data-testid="title"
    >
      Canada
    </h2>
    <p
      data-testid="textContent"
    >
      Shop for Photos!
    </p>
  </div>
</DocumentFragment>
`;
